tests/unit/jobserver/views/test_workspaces.py

```diff
@@ -488,6 +484,37 @@ def get_repo_is_private(self, owner, name):
     assert response.context_data["show_publish_repo_warning"]
 
 
+def test_workspacedetail_authorized_private_repo_show_workspace_admin_panel(
+    rf, project_membership, role_factory
+):
+    project = ProjectFactory()
+
+    # a workspace with a "private" repo
+    workspace = WorkspaceFactory(
+        project=project, repo=RepoFactory(url="http://example.com/repo/private1")
+    )
+
+    user = UserFactory()
+    BackendMembershipFactory(user=user)
+    project_membership(
+        project=project,
+        user=user,
+        roles=[role_factory(permission=permissions.workspace_archive)],
```

Comment:
nit: Another possible condition that will result in `show_admin` is `permissions.workspace_toggle_notifications` being enabled, which might also be worth testing.